src/components/scenes/PasswordLoginScene.tsx

```diff
@@ -224,19 +224,19 @@ export const PasswordLoginScene = (props: Props) => {
         const result = await Airship.show<boolean | undefined>(bridge => (
           <ChallengeModal bridge={bridge} challengeError={challengeError} />
         ))
-        if (result === true) {
+        if (result == null) return // User closed the modal
+        if (result) {
           // Try again with the passed challenge ID included
           await handleSubmit(challengeError.challengeId)
         } else {
           setPasswordErrorMessage(lstrings.failed_captcha_error)
         }
-        return
+      } else {
```

Comment:
Remove the `else`, and just do a `return`. That way this code block works like all its siblings.